tools/arriba.wdl

```diff
@@ -0,0 +1,325 @@
+## [Homepage](https://arriba.readthedocs.io/en/latest/)
+version 1.1
+
+task arriba {
+    meta {
+        description: "Run Arriba structural variant caller on a RNA-Seq BAM file."
+        help: "Typical input is a STAR-aligned BAM. Arriba also supports DRAGEN-aligned BAMs and any spec compliant BAM. That is discordant mates must have `BAM_FPROPER_PAIR (0x2)`, split reads must have `BAM_FSUPPLEMENTARY (0x800)`, and the anchor read must have a `SA` tag. Arriba also uses the `HI` tag to group supplementary alignments."
+        outputs: {
+            fusions: "Output file of fusions in TSV format"
+            discarded_fusions: "Output file of discarded fusions in TSV format"
+        }
+    }
+
+    parameter_meta {
+        bam: "Input BAM format file from which to call fusions"
+        gtf: "GTF features file. Gzipped or uncompressed."
+        reference_fasta_gz: "Gzipped reference genome in FASTA format"
+        chimeric_sam: "Optional input file of chimeric reads in SAM format, from older versions of STAR"
+        exclude_list: {
+            description: "Optional input file of regions to exclude from analysis in tab delimited format"
+            external_help: "https://arriba.readthedocs.io/en/latest/input-files/#blacklist"
+        }
+        known_fusions: {
+            description: "Optional input file of known fusions in tab delimited format"
+            extenral_help: "https://arriba.readthedocs.io/en/latest/input-files/#known-fusions"
+        }
+        annotate_fusions: {
+            description: "Optional input file in tab delimited format of fusions to annotate with tags"
+            external_help: "https://arriba.readthedocs.io/en/latest/input-files/#tags"
+        }
+        protein_domains: {
+            description: "Optional input file of protein domains coordinates in GFF3 format"
+            external_help: "https://arriba.readthedocs.io/en/latest/input-files/#protein-domains"
+        }
+        wgs_svs: {
+            description: "Optional input file of structural variants found by WGS in tab delimited or VCF format"
+            external_help: "https://arriba.readthedocs.io/en/latest/input-files/#structural-variant-calls-from-wgs"
+        }
+        interesting_contigs: "Array of contigs to consider for analysis."
+        viral_contigs: "Array of contigs to consider for viral integation site analysis."
+        disable_filters: "Array of filters to disable."
+        feature_name: "List of feature names to use in GTF."
+        strandedness: {
+            description: "Strandedness of the input data."
+            external_help: "https://arriba.readthedocs.io/en/latest/command-line-options/"
+            choices: [
+                "auto",
+                "yes",
+                "no",
+                "reverse"
+            ]
+        }
+        mark_duplicates: "Mark duplicates in the input BAM file with Arriba."
```

Comment:
When this is enabled, does it create an output BAM? If so, that should be output. If not, I don't understand this parameter.

Reply:
No, I think it computes duplicate reads on-the-fly when examining a potential fusion site.

```
-u : Arriba performs marking of duplicates internally based on identical mapping coordinates. When this switch is set, internal marking of duplicates is disabled and Arriba assumes that duplicates have been marked by a preceding program. In this case, Arriba only discards alignments flagged with the BAM_FDUP flag. This makes sense when duplicates cannot be reliably identified solely based on their mapping coordinates, e.g. when unique molecular identifiers (UMIs) are used or when independently generated libraries are merged in a single BAM file and the read group must be interrogated to distinguish duplicates from reads that map to the same coordinates by chance. In addition, when this switch is set, duplicate reads are not considered for the calculation of the coverage at fusion breakpoints (columns coverage1 and coverage2 in the output file).
```